File: emp_ui/lib/models/training.dart
class TrainingModel {
  final int? id;
  final String trainingTitle;
  final String trainerName;
  final String description;
  final String startDate;
  final String endDate;
  final String location;
  final String certificateNumber;
  final String issueDate;
  final String grade;

  TrainingModel({
    this.id,
    required this.trainingTitle,
    required this.trainerName,
    required this.description,
    required this.startDate,
    required this.endDate,
    required this.location,
    required this.certificateNumber,
    required this.issueDate,
    required this.grade,
  });

  factory TrainingModel.fromJson(Map<String, dynamic> json) {
    return TrainingModel(
      id: json['id'],
      trainingTitle: json['trainingTitle'] ?? '',
      trainerName: json['trainerName'] ?? '',
      description: json['description'] ?? '',
      startDate: json['startDate'] ?? '',
      endDate: json['endDate'] ?? '',
      location: json['location'] ?? '',
      certificateNumber: json['certificateNumber'] ?? '',
      issueDate: json['issueDate'] ?? '',
      grade: json['grade'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "trainingTitle": trainingTitle,
      "trainerName": trainerName,
      "description": description,
      "startDate": startDate,
      "endDate": endDate,
      "location": location,
      "certificateNumber": certificateNumber,
      "issueDate": issueDate,
      "grade": grade,
    };
  }
}

File: emp-api/src/main/java/com/example/entities/Training.java
package com.example.entities;

import jakarta.persistence.*;

import java.time.LocalDate;

@Entity
@Table(name = "training_management")
public class Training {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // Training Info
    private String trainingTitle;
    private String trainerName;

    @Column(length = 1000)
    private String description;

    private LocalDate startDate;
    private LocalDate endDate;

    private String location;
    private String certificateNumber;
    private LocalDate issueDate;
    private String grade;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTrainingTitle() {
        return trainingTitle;
    }

    public void setTrainingTitle(String trainingTitle) {
        this.trainingTitle = trainingTitle;
    }

    public String getTrainerName() {
        return trainerName;
    }

    public void setTrainerName(String trainerName) {
        this.trainerName = trainerName;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public LocalDate getStartDate() {
        return startDate;
    }

    public void setStartDate(LocalDate startDate) {
        this.startDate = startDate;
    }

    public LocalDate getEndDate() {
        return endDate;
    }

    public void setEndDate(LocalDate endDate) {
        this.endDate = endDate;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public String getCertificateNumber() {
        return certificateNumber;
    }

    public void setCertificateNumber(String certificateNumber) {
        this.certificateNumber = certificateNumber;
    }

    public LocalDate getIssueDate() {
        return issueDate;
    }

    public void setIssueDate(LocalDate issueDate) {
        this.issueDate = issueDate;
    }

    public String getGrade() {
        return grade;
    }

    public void setGrade(String grade) {
        this.grade = grade;
    }
}
